packages/react-components/src/components/Picker/PickerList.tsx

```diff
@@ -0,0 +1,219 @@
+import * as React from 'react';
+import cx from 'clsx';
+import { Check } from '@livechat/design-system-icons/react/material';
+import { Icon, IconTypeName } from '../Icon';
+import styles from './PickerList.module.scss';
+import { KeyCodes } from '../../utils/keyCodes';
+import { TriggerSize } from './Trigger';
+
+const baseClass = 'picker-list';
+const itemClassName = `${baseClass}__item`;
+
+export interface IPickerListItem {
+  key: string;
+  name: string;
+  groupHeader?: boolean;
+  disabled?: boolean;
+}
+
+export interface IPickerListProps {
+  isOpen: boolean;
+  items: IPickerListItem[];
+  selectedItem: IPickerListItem | null;
+  size?: TriggerSize;
+  emptyStateText?: string;
+  onClose: () => void;
+  onSelect: (item: IPickerListItem) => void;
+}
+
+export const PickerList: React.FC<IPickerListProps> = ({
+  isOpen,
+  items,
+  selectedItem,
+  size = 'medium',
+  emptyStateText = 'No results found',
+  onClose,
+  onSelect,
+}) => {
+  const mergedClassNames = cx(
+    styles[baseClass],
+    styles[`${baseClass}--${size}`],
+    {
+      [styles[`${baseClass}__no-results`]]: items.length === 0,
+    }
+  );
+
+  const [currentItemKey, setCurrentItemKey] = React.useState<string | null>(
+    null
+  );
+  const indexRef = React.useRef(-1);
+  const lastIndexRef = React.useRef(0);
+  const listRef = React.useRef<HTMLUListElement>(null);
+
+  React.useEffect(() => {
+    indexRef.current = 0;
+
+    if (items.length > 0) {
+      setCurrentItemKey(items[indexRef.current].key);
+    }
+
+    if (isOpen) {
+      document.addEventListener('keydown', onKeyDown);
+    }
+
+    return () => document.removeEventListener('keydown', onKeyDown);
+  }, [items, isOpen]);
+
+  const scrollItems = () => {
+    if (!listRef.current) {
+      return;
+    }
+
+    const focusedElement = listRef.current.querySelector(
+      `[aria-current="true"]`
+    );
+
+    if (focusedElement instanceof HTMLElement) {
+      const { height: ulHeight, top: ulTop } =
+        listRef.current.getBoundingClientRect();
+
+      const { height: itemHeigth, top: itemTop } =
+        focusedElement.getBoundingClientRect();
+
+      const relativeTop = itemTop + itemHeigth - ulTop;
+      const itemOfsetTop = focusedElement.offsetTop;
+
+      if (relativeTop > ulHeight) {
+        listRef.current.scrollTop = itemOfsetTop - ulHeight + itemHeigth;
+      } else if (itemTop < ulTop) {
+        listRef.current.scrollTop = itemOfsetTop - (itemOfsetTop % itemHeigth);
+      }
+    }
+  };
```

Comment:
This function and all of those calculations can be replaced using [`scrollIntoView`](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) on the currently active element.
We can use the `ref` prop from "host" elements which can receive a function that on each render would get the reference to the very specfifc DOM node of this element. Combining this with `currentItemKey` state value we can decide on which element to call `scrollIntoView`. That way we will receive exactly the same behavior and user experience without doing all calculations by hand 🙂 

It will look something like this:
```jsx
<li
  ref={element => {
    if(currentItemKey === item.key) {
      element?.scrollIntoView({ block: 'nearest' })
    }
  }}
>
{item.name}
</li>
```

```suggestion
```